src/devices/ubisys.ts

```diff
@@ -815,11 +815,38 @@ const definitions: Definition[] = [
         toZigbee: [tz.cover_state, tz.cover_position_tilt, tz.metering_power,
             ubisys.tz.configure_j1, ubisys.tz.configure_device_setup,
             tz.currentsummdelivered],
-        exposes: [
-            e.cover_position_tilt(),
-            e.power().withAccess(ea.STATE_GET),
-            e.energy().withAccess(ea.STATE_GET),
-        ],
+        exposes: (device, options) => {
+            const coverExpose = e.cover();
+            if (device) {
+                const coverType = device.getEndpoint(1).getClusterAttributeValue('closuresWindowCovering', 'windowCoveringType');
+                switch (coverType) { // cf. Ubisys J1 Technical Reference Manual, chapter 7.2.5.1 Calibration
+                case 0: // Roller Shade, Lift only
+                case 1: // Roller Shade two motors, Lift only
+                case 2: // Roller Shade exterior, Lift only
+                case 3: // Roller Shade two motors exterior, Lift only
+                case 4: // Drapery, Lift only
+                case 5: // Awning, Lift only
+                case 9: // Projector Screen, Lift only
+                    coverExpose.withPosition();
+                    break;
+                case 6: // Shutter, Tilt only
+                case 7: // Tilt Blind, Tilt only
+                    coverExpose.withTilt();
+                    break;
+                case 8: // Tilt Blind, Lift & Tilt
+                default:
+                    coverExpose.withPosition().withTilt();
+                    break;
+                }
+            } else {
+                coverExpose.withPosition().withTilt();
+            }
+            return [
+                coverExpose,
+                e.power().withAccess(ea.STATE_GET),
+                e.energy().withAccess(ea.STATE_GET),
+                e.linkquality()];
```

Comment:
For readability you might want to move the `]` to it's own line.

Reply:
Might need to push another lint fix as you also need a `,` after `e.linkquality()`

Reply:
yes, just noticed that, too. thanks :-)